### PR TITLE
chore: add some Web Platform Tests testcases

### PR DESCRIPTION
**Pull request by drwpow**
## Summary

Read #7128 for more context. But the [Web Platform Tests](https://github.com/web-platform-tests/) have hundreds of test cases that could be brought into Biome to capture nuanced browser behavior when it comes to ARIA role calculation. It really shines with edge cases, and highlights all the ways Biome would deviate from browser behavior.

This PR doesn’t introduce any changes! It only adds more test-cases. Looking for feedback on where these live, because there’ll be a lot more of these 🙏 

### Alternatives

What we’re really doing is **calculating roles**, which there aren’t unit tests for. Rather than using `noRedundantRoles`’ tests, we could do something lower-level if there ends up being too much indirection here

## Test Plan

- Tests added
## Docs

Just tests?

### CODE REVIEW

File: crates/biome_js_analyze/tests/specs/a11y/noRedundantRoles/invalidHtmlAamRoleGeneric.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+<>
+	<b role="generic">x</b>
+	<bdi role="generic">x</bdi>
+	<bdo role="generic">x</bdo>
+	<data value="1" role="generic">
+		x
+	</data>
+	<div open role="generic">
+		x
+	</div>
```
[the remark this answers is not in the document]
Comment (drwpow) [reply crates/biome_js_analyze/tests/specs/a11y/noRedundantRoles/invalidHtmlAamRoleGeneric.jsx:11]: Testing invalid HTML is intentionally part of this test

File: crates/biome_js_analyze/tests/specs/a11y/noRedundantRoles/invalidHtmlAamRoleGeneric.jsx
Comment (dyc3) [inline crates/biome_js_analyze/tests/specs/a11y/noRedundantRoles/invalidHtmlAamRoleGeneric.jsx:1]: Perhaps we should put imported tests into their own folder.
